### PR TITLE
feat(feedback): update inputs with react hook form

### PR DESCRIPTION
**Pull request by ardasnturk**
- updated the inputs on the `FeedbackScreen` with `react-hook-form` to have the same structure everywhere in the code
- added `useMutation` instead of the old `Mutation` component to be able to use it with new inputs
- added all informative texts to `texts` file 
- changed styling to match the pre-update design
- added design missing props to `Input.tsx`
- added `boldLabel` feature to `Input`
- added `boldTitle` feature to `Checkbox`
- added `disabled` feature to send button on `FeedbackScreen`
- added `*` next to the labels of required inputs that must be filled

SVA-911

‼️ after this update, we can show the error messages we have created for the users under the inputs ‼️
‼️ before the update, we were only showing the please fill in the missing fields alert ‼️

* [x] Android portrait mode
* [x] iOS portrait mode
* [x] Android landscape mode
* [x] iOS landscape mode

## Screenshots:

|before|after|
|--|--|
Simulator Screen Shot - iPhone 14 - 2023-02-02 at 15 33 40 | Simulator Screen Shot - iPhone 14 - 2023-02-02 at 15 33 55

|before|after|
|--|--|
Simulator Screen Shot - iPhone 14 - 2023-02-02 at 15 27 54 | Simulator Screen Shot - iPhone 14 - 2023-02-02 at 15 27 34

|before|after|
|--|--|
Simulator Screen Shot - iPhone 14 - 2023-02-02 at 15 30 37 | Simulator Screen Shot - iPhone 14 - 2023-02-02 at 15 30 53

### CODE REVIEW

File: src/screens/FeedbackScreen.js [hunk 1]
```diff
@@ -1,141 +1,183 @@
...
+                control={control}
+                containerStyle={styles.containerStyle}
+              />
+
+              <Input
+                name="message"
+                label={texts.feedbackScreen.inputsLabel.message + ' *'}
+                boldLabel
+                placeholder={texts.feedbackScreen.inputsLabel.message}
+                keyboardType="phone-pad"
```
Comment (donni106) [inline src/screens/FeedbackScreen.js:133]: message shouldn't be a phone-pad, right?

File: src/screens/FeedbackScreen.js [hunk 2]
```diff
@@ -1,141 +1,183 @@
...
+                      uncheckedIcon="square-o"
+                      uncheckedColor={colors.darkText}
+                      checked={value}
+                      onPress={() => onChange(!value)}
+                    />
+                    <Input
+                      control={control}
+                      errorMessage={errors.consent && errors.consent.message}
+                      hidden
+                      name={'consent'}
```
Comment (donni106) [inline src/screens/FeedbackScreen.js:163]: `name="consent"` could be a string, right?

File: src/screens/FeedbackScreen.js [hunk 3]
```diff
@@ -1,141 +1,183 @@
...
+                      validate
+                    />
```
Comment (donni106) [inline src/screens/FeedbackScreen.js:165]: please adjust the error text for the checkbox with less padding from the checkbox and label

image

there should be more padding to the button compared to the checkbox and label

**Review by donni106 — CHANGES_REQUESTED** [review]

File: src/screens/FeedbackScreen.js [hunk 4]
```diff
@@ -1,141 +1,183 @@
...
+                    <Checkbox
+                      boldTitle
+                      title={texts.feedbackScreen.inputsLabel.checkbox + ' *'}
+                      checkedIcon="check-square-o"
+                      checkedColor={colors.accent}
+                      uncheckedIcon="square-o"
+                      uncheckedColor={colors.darkText}
+                      checked={value}
+                      onPress={() => onChange(!value)}
+                    />
```
Comment (donni106) [inline src/screens/FeedbackScreen.js:158]: after submission, everything except the checkbox is resetted

image

please see that the checkbox value gets resetted as well

File: src/components/form/Input.tsx [hunk 5]
```diff
@@ -79,15 +85,15 @@ export const Input = ({
   return (
     <RNEInput
       ref={inputRef}
-      label={<Label>{label}</Label>}
+      label={boldLabel ? <BoldText>{label}</BoldText> : <Label>{label}</Label>}
```
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/components/form/Input.tsx]: Yes, very true. I'm making the necessary changes. 👍

https://github.com/smart-village-solutions/smart-village-app-app/pull/741/commits/6837d2168f5ea1051c993a9d1976da89fcd1dd85

File: src/components/form/Input.tsx [hunk 6]
```diff
@@ -79,15 +85,15 @@ export const Input = ({
...
       value={field.value}
       onChangeText={field.onChange}
       onBlur={field.onBlur}
       disabled={disabled}
       multiline={multiline}
       {...furtherProps}
       errorMessage={!isValid ? errorMessage : ''}
-      scrollEnabled={multiline && false}
+      scrollEnabled={multiline}
```
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/components/form/Input.tsx:96]: `FeedbackScreen` was using `react-native`'s `TextInput` before this update. Therefore, multiline message input could be scrolled.

But now we use our own Input to display error messages and the code `multiline && false` prevents the message input from scrolling. 

I removed the `&& false` value and made the input scrollable again.

File: src/components/form/Input.tsx [hunk 7]
```diff
@@ -99,20 +105,23 @@ export const Input = ({
       containerStyle={[
         styles.container,
         row && styles.row,
-        hidden && !errorMessage && styles.containerHidden
+        hidden && !errorMessage && styles.containerHidden,
+        !!containerStyle && containerStyle
```
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/components/form/Input.tsx]: Yes, you are right. I'm updating.

https://github.com/smart-village-solutions/smart-village-app-app/pull/741/commits/fc0e103e7cf075cc795ac910df21b6f7e7a63805

File: src/components/form/Input.tsx [hunk 8]
```diff
@@ -99,20 +105,23 @@ export const Input = ({
...
       ]}
       inputContainerStyle={[
         styles.inputContainer,
         disabled && styles.inputContainerDisabled,
         hidden && styles.inputContainerHidden,
         isValid && styles.inputContainerSuccess,
-        !isValid && !!errorMessage && styles.inputContainerError
+        !isValid && !!errorMessage && styles.inputContainerError,
+        !!inputContainerStyle && inputContainerStyle
```
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/components/form/Input.tsx]: Yes, you are right. I will update. 👍

https://github.com/smart-village-solutions/smart-village-app-app/pull/741/commits/fc0e103e7cf075cc795ac910df21b6f7e7a63805

File: src/components/form/Input.tsx [hunk 9]
```diff
@@ -99,20 +105,23 @@ export const Input = ({
...
       ]}
       rightIconContainerStyle={styles.rightIconContainer}
       inputStyle={[
         styles.input,
         multiline && device.platform === 'ios' && styles.multiline,
-        !isValid && !!errorMessage && styles.inputError
+        !isValid && !!errorMessage && styles.inputError,
+        !!inputStyle && inputStyle
```
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/components/form/Input.tsx]: Yes, you are right. I will update. 👍

https://github.com/smart-village-solutions/smart-village-app-app/pull/741/commits/fc0e103e7cf075cc795ac910df21b6f7e7a63805

File: src/screens/FeedbackScreen.js (hunk 3 above)
Comment (ardasnturk) [reply src/screens/FeedbackScreen.js]: the problem is that the input is hidden. even though it is hidden, a space is automatically created for the `label`.
solving the problem.

https://github.com/smart-village-solutions/smart-village-app-app/pull/741/commits/b8634cc9c32965eae9f7f95d7dff459a2e310ed9

File: src/screens/FeedbackScreen.js [hunk 10]
```diff
@@ -1,141 +1,183 @@
...
-        email,
-        phone,
-        message,
-        consent,
+        name: createAppUserContentNewData.name,
+        email: createAppUserContentNewData.email,
+        phone: createAppUserContentNewData.phone,
+        message: createAppUserContentNewData.message,
+        consent: createAppUserContentNewData.consent,
         appInfo: appInfo
```
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/screens/FeedbackScreen.js]: https://github.com/smart-village-solutions/smart-village-app-app/pull/741/commits/b8634cc9c32965eae9f7f95d7dff459a2e310ed9

File: src/screens/FeedbackScreen.js (hunk 4 above)
Comment (ardasnturk) [reply src/screens/FeedbackScreen.js]: The hidden input we created prevents the checkbox from resetting for some reason. So I suggest to remove the hidden input and raise an alert if the checkbox is unchecked. 

https://github.com/smart-village-solutions/smart-village-app-app/pull/741/commits/d28a8ed30bf4698775757852e5230ba93c8bdde4

File: src/screens/FeedbackScreen.js (hunk 2 above)
Comment (ardasnturk) [reply src/screens/FeedbackScreen.js]: I don't think this change is necessary as the hidden input has been deleted.

https://github.com/smart-village-solutions/smart-village-app-app/pull/741/commits/d28a8ed30bf4698775757852e5230ba93c8bdde4

File: src/screens/FeedbackScreen.js [hunk 11]
```diff
@@ -1,141 +1,183 @@
...
+                keyboardType="phone-pad"
+                multiline
+                textAlignVertical="top"
+                validate
+                rules={{ required: texts.feedbackScreen.inputsErrorMessages.message }}
+                errorMessage={errors.message && errors.message.message}
+                control={control}
+                containerStyle={styles.containerStyle}
+                inputStyle={styles.textArea}
+              />
```
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/screens/FeedbackScreen.js]: It's a stylistic problem. I'll fix it.

https://github.com/smart-village-solutions/smart-village-app-app/pull/741/commits/b8634cc9c32965eae9f7f95d7dff459a2e310ed9

[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/screens/FeedbackScreen.js]: Sorry, I forgot to upload a picture.

|||
|--|--|
image | image

File: src/screens/FeedbackScreen.js [hunk 12]
```diff
@@ -1,141 +1,179 @@
...
-    return consent;
-  };
+  const onSubmit = async (createAppUserContentNewData) => {
+    Keyboard.dismiss();
+
+    if (!createAppUserContentNewData.consent) {
+      return Alert.alert(
+        texts.feedbackScreen.inputsErrorMessages.hint,
+        texts.feedbackScreen.inputsErrorMessages.checkbox
+      );
```
[the remark this answers is not in the document]
Comment (ardasnturk) [reply src/screens/FeedbackScreen.js:52]: yes. if we can't assign an extra onPress to the alert's button, we can ignore this feature.

I will update Alert in the rest of the code.

|before|after|
|--|--|
image | image

the feedback form will be reset before the alert button is pressed.

https://github.com/smart-village-solutions/smart-village-app-app/commit/9874c7548e76006693a8757802649df15f133a1c

[the remark this answers is not in the document]
Comment (donni106) [reply src/screens/FeedbackScreen.js:52]: and the alert shows the same title, message and buttons compared to before?

**Review by donni106 — APPROVED** [review]

**Pull request merged**